src/client.ts

```diff
@@ -0,0 +1,92 @@
+import { PoolConfig } from 'pg';
+import { PostgresSystemDatabase, SystemDatabase, WorkflowStatusInternal } from './system_database';
+import { GlobalLogger as Logger } from './telemetry/logs';
+import { v4 as uuidv4 } from 'uuid';
+import { RetrievedHandle, StatusString, WorkflowHandle } from './workflow';
+
+interface EnqueueOptions {
+  queueName: string;
+  workflowName: string;
+  workflowClassName: string;
+  workflowID?: string;
+  maxRetries?: number;
+  appVersion?: string;
+}
+
+export class DBOSClient {
+  private readonly logger: Logger;
+  private readonly systemDatabase: SystemDatabase;
+
+  constructor(poolConfig: PoolConfig, systemDatabase: string) {
+    this.logger = new Logger();
+    this.systemDatabase = new PostgresSystemDatabase(poolConfig, systemDatabase, this.logger);
+  }
+
+  async init() {
+    await this.systemDatabase.init();
+  }
+
+  async destroy() {
+    await this.systemDatabase.destroy();
+  }
+
+  async enqueue<T extends unknown[]>(options: EnqueueOptions, ...args: T): Promise<void> {
+    const { workflowName, workflowClassName, queueName, appVersion } = options;
+    const workflowUUID = options.workflowID ?? uuidv4();
+    const maxRetries = options.maxRetries ?? 50;
+
+    const internalStatus: WorkflowStatusInternal = {
+      workflowUUID: workflowUUID,
+      status: StatusString.ENQUEUED,
+      workflowName: workflowName,
+      workflowClassName: workflowClassName,
+      workflowConfigName: '',
+      queueName: queueName,
+      authenticatedUser: '',
+      output: undefined,
+      error: '',
+      assumedRole: '',
+      authenticatedRoles: [],
+      request: {},
+      executorId: '',
+      applicationVersion: appVersion,
+      applicationID: '',
+      createdAt: Date.now(),
+      maxRetries: maxRetries ?? 50,
+    };
+
+    await this.systemDatabase.initWorkflowStatus(internalStatus, args);
+    await this.systemDatabase.enqueueWorkflow(workflowUUID, queueName);
+  }
+
+  async send<T>(destinationID: string, message: T, topic?: string, idempotencyKey?: string): Promise<void> {
+    idempotencyKey ??= uuidv4();
+    const internalStatus: WorkflowStatusInternal = {
+      workflowUUID: `${destinationID}-${idempotencyKey}`,
+      status: StatusString.SUCCESS,
```

Comment:
As @maxdml mentioned below, if we set the status to `SUCCESS` here, then if the actual send failed, the user would mistakenly think the send has succeeded (showed up as `SUCCESS` on our console). One solution is to init this as `PENDING` and record the result as `SUCCESS` later.

Reply:
That's a mess, as then things might try to recover it. It's honestly fine as is, and we absolutely should not add more complexity here.

Reply:
As I understand it, the point of this enqueue is to work directly against the system DB. If we wanted to expose a mechanism that was as reliable as the running DBOS app, we should expose an enqueue endpoint in the admin service since it has access to the executor. Alternatively, we could ask customers to build their own HTTP endpoint that would call `startWorkflow` and set the `workflowID` and `queueName` as they see fit

Reply:
This comment is for `send` not enqueue. I think the enqueue implementation is fine as is.
For `send`, as @kraftp explained, we probably don't need to record the send in the workflow status table.

Reply:
Sorry for mixing up the methods, but my point still stands. We still don't have an executor in the client that would be needed for reliable send. The points about exposing an admin API and/or a customer endpoint are still valid approaches. 

Alternatively, we could reliably send by doing the work of `initWorkflowStatus` and `send` inside a transaction, but that would be either a significant refactoring of system db or a lot of copy/paste.